### PR TITLE
feat(erc20): Support fee-on-transfer ERC20 tokens

### PR DESCRIPTION
**Pull request by calbera**

### CODE REVIEW

**Comment by itsdevbear** [comment]
@calbera is this getting in?

**Comment by itsdevbear** [comment]
this looks fine, just all the tests are commented out.

File: cosmos/precompile/erc20/middleware.go [hunk 1]
```diff
@@ -180,18 +182,48 @@ func (c *Contract) transferERC20ToCoin(
...
+			c.RegistryKey(), token, c.polarisERC20ABI,
+			balanceOf, c.RegistryKey(),
+		)
+		if err != nil {
+			return err
+		}
+		balanceAfter = utils.MustGetAs[*big.Int](ret[0])
+
 		// mint amount Polaris Coins to recipient
+		amount = new(big.Int).Sub(balanceAfter, balanceBefore)
```
Comment (calbera) [inline cosmos/precompile/erc20/middleware.go:226]: @Xenomega

**Comment by calbera** [comment]
so this should be good 2 go now @itsdevbear

**Review by itsdevbear — APPROVED** [review]

**Pull request merged**